wal_test.go

```diff
@@ -265,176 +145,6 @@ func TestSegmentWAL_Log_Restore(t *testing.T) {
 	}
 }
 
-func TestWALRestoreCorrupted_invalidSegment(t *testing.T) {
```

Comment:
Hate to say this but these two tests (this and one after) also need to be supported :/ Other than that, LGTM!